File: src/main/java/com/matag/game/cardinstance/modifiers/CardModifiersUntilEndOfTurn.java
package com.matag.game.cardinstance.modifiers;

import java.util.ArrayList;
import java.util.List;

import com.matag.cards.properties.PowerToughness;
import com.matag.game.cardinstance.ability.CardInstanceAbility;

import lombok.Data;

@Data
public class CardModifiersUntilEndOfTurn {
  private boolean attacked;
  private boolean blocked;
  private List<CardInstanceAbility> extraAbilities = new ArrayList<>();
  private PowerToughness extraPowerToughness = new PowerToughness(0, 0);
  private String newController;
  private boolean toBeDestroyed;
  private boolean toBeReturnedToHand;

  public void addExtraPowerToughnessUntilEndOfTurn(PowerToughness extraPowerToughness) {
    var newPower = this.extraPowerToughness.getPower() + extraPowerToughness.getPower();
    var newToughness = this.extraPowerToughness.getToughness() + extraPowerToughness.getToughness();
    this.extraPowerToughness = new PowerToughness(newPower, newToughness);
  }
}

File: src/main/kotlin/com/matag/game/cardinstance/modifiers/CardModifiersUntilEndOfTurn.kt
package com.matag.game.cardinstance.modifiers

import com.matag.cards.ability.Ability
import com.matag.cards.properties.PowerToughness

class CardModifiersUntilEndOfTurn {
    val isAttacked = false
    var isBlocked = false
    var extraAbilities: List<Ability> = listOf()
    var extraPowerToughness = PowerToughness(0, 0)
    var newController: String? = null
    var isToBeDestroyed = false
    var isToBeReturnedToHand = false

    fun addExtraPowerToughnessUntilEndOfTurn(extraPowerToughness: PowerToughness) {
        val newPower = this.extraPowerToughness.power + extraPowerToughness.power
        val newToughness = this.extraPowerToughness.toughness + extraPowerToughness.toughness
        this.extraPowerToughness = PowerToughness(newPower, newToughness)
    }
}
